crates/swift-bridge-ir/src/parse/parse_enum/enum_attributes.rs

```diff
@@ -0,0 +1,125 @@
+use crate::bridged_type::DeriveAttrs;
+use crate::errors::ParseError;
+use crate::parse::move_input_cursor_to_next_comma;
+use proc_macro2::Ident;
+use quote::ToTokens;
+use syn::parse::{Parse, ParseStream};
+use syn::{Attribute, LitStr, Path, Token};
+
+#[derive(Default)]
+pub(super) struct SharedEnumAllAttributes {
+    pub errors: Vec<ParseError>,
+    pub swift_bridge: SharedEnumSwiftBridgeAttributes,
+    pub derive: DeriveAttrs,
+}
+
+impl SharedEnumAllAttributes {
+    pub(super) fn from_attributes(attribs: &[Attribute]) -> Result<Self, syn::Error> {
+        let mut attributes = SharedEnumAllAttributes::default();
+
+        for attr in attribs {
+            let attribute_name = attr.path.to_token_stream().to_string();
+
+            match attribute_name.as_str() {
+                "derive" => {
+                    let parser =
+                        syn::punctuated::Punctuated::<Path, syn::Token![,]>::parse_terminated;
+                    let args = attr.parse_args_with(parser)?;
+                    for arg in args.into_iter() {
+                        match arg.get_ident().unwrap().to_string().as_str() {
+                            "Debug" => attributes.derive.debug = true,
+                            _ => todo!("Unsupported derive macro; please see https://github.com/chinedufn/swift-bridge/issues/190#issuecomment-1463234027"),
+                        }
+                    }
+                }
+                "swift_bridge" => {
+                    attributes.swift_bridge = attr.parse_args()?;
+                    // note: this will empty attributes.swift_bridge.errors, maybe it would be better to clone?
+                    attributes
+                        .errors
+                        .append(&mut attributes.swift_bridge.errors);
```

Comment:
Can delete the comment and leave it as is for now. 

I say this because if we commented on every possible trivial improvement the codebase might have hundreds of these. 
I think it's fine to expect a future maintainer to know how to refactor this if they want to make things more correct.

Nevertheless I appreciate your drive for correctness!

---

Alternatively, you could make an `AttrParseResult` enum type and do something like:

```rust
match attr.parse_args::<AttrParseResult<SharedEnumSwiftBridgeAttributes>>()? {
    AttrParseResult::Ok(ok) => attributes.swift_bridge = ok,
    AttrParseResult::Err(errors) => attributes.errors.append(&mut errors)
}
```

This way it becomes clear that there are either attributes, or errors, and there isn't the chance to check for `.errors` and see nothing since we drained them.

---

I'm fine with either approach since this is all fairly self-contained internal stuff.